app/enterprise.go

```diff
@@ -136,7 +142,13 @@ func (s *Server) initEnterprise() {
 		s.Notification = notificationInterface(s.FakeApp())
 	}
 	if samlInterface != nil {
-		s.Saml = samlInterface(s.FakeApp())
+		if *s.FakeApp().Config().ExperimentalSettings.UseNewSAMLLibrary && samlInterfaceNew != nil {
+			mlog.Debug("Loading new SAML2 library")
+			s.Saml = samlInterfaceNew(s.FakeApp())
+		} else {
+			mlog.Debug("Loading original SAML library")
```

Comment:
Might want to log a message if the new saml library was configured, but not registered (i.e. not licensed), in case someone tries this from team edition?